### PR TITLE
Update Dockerfile handling to keep parser directives at the top

### PR DESCRIPTION
**Pull request by crazy-max**
Handle Dockerfile `# syntax` directives.

Atm the syntax is quite lax but we plan to make it stricter in the future (https://docs.docker.com/engine/reference/builder/#parser-directives).

cc @thaJeztah

Signed-off-by: CrazyMax <google@crazymax.dev>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -296,6 +296,7 @@ var head = []string{
 	"# encoding:",              // Ruby encoding
 	"# frozen_string_literal:", // Ruby interpreter instruction
 	"<?php",                    // PHP opening tag
+	"# syntax=",                // Dockerfile directive https://docs.docker.com/engine/reference/builder/#parser-directives
```
Comment (thaJeztah) [inline main.go:299]: Perhaps use `# syntax` (without `=`) to allow for spaces around the `=`
Also may want to add `# escape`, which is also a directive.

Comment (crazy-max) [reply main.go:299]: > 
> 
> Perhaps use `# syntax` (without `=`) to allow for spaces around the `=`
> Also may want to add `# escape`, which is also a directive.

:+1:

> 🤔 wondering what it does if a Dockerfile has _both_ (`syntax` _and_ `escape`), although `escape` is mostly used on Windows, so not yet using BuildKit (and thus no `syntax`)

Not sure actually good point.

**Comment by willnorris** [comment]
@crazy-max, I just checked in some more focused testing methods... would you mind rebasing on master, and add [test cases here](https://github.com/google/addlicense/blob/058aeeb7be096c4bcde8a2c0ceca2f6584fb5711/main_test.go#L236-L244)?  I'm planning to eventually move us away from all of the individual files in `testdata/**`, so you can leave those out.

**Comment by crazy-max** [comment]
@willnorris  Done

**Review by willnorris — APPROVED** [review]

**Pull request merged**